src/net/mod.rs

```diff
@@ -2195,41 +2198,78 @@ pub mod test {
             )
             .unwrap();
 
-            let init_code = config.setup_code.clone();
-            let (chainstate, _) = StacksChainState::open_and_exec(false, config.network_id, &chainstate_path, Some(config.initial_balances.clone()),
-                |ref mut clarity_tx| {
-                    if init_code.len() > 0 {
-                        clarity_tx.connection().as_transaction(|clarity| {
-                            let boot_code_address = StacksAddress::from_string(&STACKS_BOOT_CODE_CONTRACT_ADDRESS.to_string()).unwrap();
-                            let boot_code_account = StacksAccount {
-                                principal: PrincipalData::Standard(StandardPrincipalData::from(boot_code_address.clone())),
-                                nonce: 0,
-                                stx_balance: STXBalance::zero(),
-                            };
-                            let boot_code_auth = TransactionAuth::Standard(TransactionSpendingCondition::Singlesig(SinglesigSpendingCondition {
+            let conf = config.clone();
+            let post_flight_callback = move |clarity_tx: &mut ClarityTx| {
+                if conf.setup_code.len() > 0 {
```

Comment:
The boot code is always going to be instantiated, regardless of `setup_code` (also, why is this no longer `init_code`?).  This conditional should be removed.